crates/common/consensus/src/merkle.rs

```diff
@@ -0,0 +1,150 @@
+use anyhow::ensure;
+use tree_hash::Hash256;
```

Comment:
we should be using `use alloy::primitives::B256` rather then Hash256, which is just a type alias to B256